fdbserver/include/fdbserver/DataDistribution.actor.h

```diff
@@ -148,6 +148,32 @@ struct RelocateShard {
 	    moveReason(DataMovementReason::INVALID) {}
 };
 
+struct GetTeamRequest;
+namespace data_distribution {
+class EligibilityCounter {
+public:
+	enum Type {
+		NONE = 0, // don't care about eligibility
+		LOW_CPU = 1,
+		LOW_DISK_UTIL = 2
+	};
+
+	// set the count of type to 0
+	void reset(Type type);
+
+	// return the minimal count of a combined eligible type
+	unsigned getCount(int combinedType) const;
+
+	// increase the count of type
+	void increase(Type type);
```

Comment:
Seems like we haven't fully used the add-up values of `increase` function. For now, is it same as setting `type_count[Type]=1`? What's the point of having a `std::map<Type, unsigned int> type_count` and sum up the count?

Reply:
In case there is demand for ordering eligible teams. It's not used now, but there's no obvious extra cost to use counter compared to bool map.